manager/src/main/java/org/openremote/manager/mqtt/AssetInterceptHandler.java

```diff
@@ -0,0 +1,190 @@
+
+    AssetInterceptHandler(AssetStorageService assetStorageService,
+                          AssetProcessingService assetProcessingService,
+                          ManagerIdentityService managerIdentityService,
+                          ManagerKeycloakIdentityProvider managerKeycloakIdentityProvider,
+                          MessageBrokerService messageBrokerService, MqttConnector mqttConnector, BiConsumer<String, AttributeEvent> attributeEventConsumer) {
+
+        this.assetStorageService = assetStorageService;
+        this.assetProcessingService = assetProcessingService;
+        this.identityService = managerIdentityService;
+        this.identityProvider = managerKeycloakIdentityProvider;
+        this.messageBrokerService = messageBrokerService;
+        this.mqttConnector = mqttConnector;
+        this.attributeEventConsumer = attributeEventConsumer;
+    }
+
+    @Override
+    public String getID() {
+        return UniqueIdentifierGenerator.generateId(AssetInterceptHandler.class.getName());
+    }
+
+    @Override
+    public Class<?>[] getInterceptedMessageTypes() {
+        return InterceptHandler.ALL_MESSAGE_TYPES;
+    }
+
+    @Override
+    public void onConnect(InterceptConnectMessage interceptConnectMessage) {
+        MqttConnector.MqttConnection connection = mqttConnector.createConnection(interceptConnectMessage.getClientID(), interceptConnectMessage.getUsername(), interceptConnectMessage.getPassword());
+        String suppliedClientSecret = new String(interceptConnectMessage.getPassword(), StandardCharsets.UTF_8);
+        connection.accessToken = identityProvider.getKeycloak().getAccessToken(connection.realm, new ClientCredentialsAuthFrom(connection.username, suppliedClientSecret)).getToken();
+    }
+
+    @Override
+    public void onDisconnect(InterceptDisconnectMessage interceptDisconnectMessage) {
+        MqttConnector.MqttConnection connection = mqttConnector.removeConnection(interceptDisconnectMessage.getClientID());
+        if (connection != null) {
+            CancelEventSubscription<AttributeEvent> cancelEventSubscription = new CancelEventSubscription<>(AttributeEvent.class);
+            messageBrokerService.getProducerTemplate().sendBodyAndHeader(ClientEventService.CLIENT_EVENT_QUEUE, cancelEventSubscription, WebsocketConstants.SESSION_KEY, connection.clientId);
+        }
+    }
+
+    @Override
+    public void onConnectionLost(InterceptConnectionLostMessage interceptConnectionLostMessage) {
+        MqttConnector.MqttConnection connection = mqttConnector.removeConnection(interceptConnectionLostMessage.getClientID());
+        if (connection != null) {
+            CancelEventSubscription<AttributeEvent> cancelEventSubscription = new CancelEventSubscription<>(AttributeEvent.class);
+            messageBrokerService.getProducerTemplate().sendBodyAndHeader(ClientEventService.CLIENT_EVENT_QUEUE, cancelEventSubscription, WebsocketConstants.SESSION_KEY, connection.clientId);
```

Comment:
Should use `EventSubscriptions:cancelAll()` to remove any subscriptions this client has made